### PR TITLE
Expose HMIS custom assessments on the client dashboard

### PR DESCRIPTION
**Pull request by eanders**
[//]: # 'remove this if PR is for a release-* branch'
# _Please squash merge this PR_

## Description

This exposes HMIS custom assessments on the client dashboard in the warehouse.  These are added as rows to the Assessments card and have a link to the assessment in the HMIS as well as a modal that shows a very basic read-only view of the assessment in the same manner as ETO assessments or other similar items have been shown in the past.

Assessments are grouped by name with the most-recent by AssessmentDate visible, previous assessments are hidden behind the +.

### Custom Assessment in HMIS
Screenshot 2024-03-21 at 6 24 02 PM

### Client Dashboard in the Warehouse
Screenshot 2024-03-21 at 6 21 31 PM

### Client Dashboard expanded
Screenshot 2024-03-21 at 6 21 23 PM

### Assessment modal in the warehouse (note it integrates some CE Assessment information at the top if a CE assessment is attached.)
Screenshot 2024-03-21 at 6 20 19 PM

## Type of change
[//]: # 'remove options that are not relevant'
- [x] New feature (adds functionality)

## Checklist before requesting review
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [x] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [x] My code follows the style guidelines of this project (rubocop)
- [x] I have updated the documentation (or not applicable)
- [x] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

File: app/views/clients/hud_assessments/show.haml [hunk 1]
```diff
@@ -20,9 +20,9 @@
         %dt Prioritization Status
         %dd= HudUtility2024.prioritization_status @assessment.PrioritizationStatus
         %dt Created
-        %dd= HudUtility2024.prioritization_status @assessment.DateCreated
+        %dd= @assessment.DateCreated
```
Comment (eanders) [inline app/views/clients/hud_assessments/show.haml:23]: Unrelated bugfix

File: drivers/hmis/app/controllers/hmis_client/assessments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+# Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+class HmisClient::AssessmentsController < ApplicationController
+  include AjaxModalRails::Controller
+  include ClientDependentControllers
+
+  before_action :require_can_view_enrollment_details_tab!
```
Comment (eanders) [inline drivers/hmis/app/controllers/hmis_client/assessments_controller.rb:11]: This is the permission currently in use for being able to drill-down on other similar assessments.  Let me know if it should be something else.

File: drivers/hmis/app/models/hmis/hud/project.rb [hunk 3]
```diff
@@ -59,6 +59,8 @@ class Hmis::Hud::Project < Hmis::Hud::Base
   has_many :services, through: :enrollments_including_wip
   has_many :custom_services, through: :enrollments_including_wip
 
+  has_one :warehouse_project, class_name: 'GrdaWarehouse::Hud::Project', foreign_key: :id, primary_key: :id
```
Comment (eanders) [inline drivers/hmis/app/models/hmis/hud/project.rb:62]: Added in addition to `.as_warehouse` to allow preloading.

File: app/views/clients/rollup/_assessments.html.haml [hunk 4]
```diff
@@ -84,6 +94,40 @@
...
+        - custom_hmis_assessments.each_with_index do |(name, (assessment, *assessments)), i|
+          - any_buttons = assessments.any?
+          - cz = "assessment_type_#{i}"
+          %tr{ class: ( 'assessment__new-type' if any_buttons )}
+            %td= link_to name, hmis_client_client_assessment_path(@client, assessment), class: 'btn btn-primary btn-muted btn-xs', data: { loads_in_pjax_modal: true }
+            %td= assessment.enrollment.project.warehouse_project.name(current_user)
+            %td= assessment.AssessmentDate
+            %td= assessment.user&.name
+            - path = "client/#{assessment.enrollment.client.id}/enrollments/#{assessment.enrollment.id}/assessments/#{assessment.id}"
+            %td= link_to 'HMIS', url_for_hmis(assessment.enrollment.data_source, path), target: :_blank
```
Comment (gigxz) [inline app/views/clients/rollup/_assessments.html.haml]: Can we add `Hmis::Hud::CustomAssessment` to this function instead? 
https://github.com/greenriver/hmis-warehouse/blob/f7d4269c21328c88f490986cd89285898805b422/app/models/grda_warehouse/data_source.rb#L556-L576

Comment (eanders) [reply app/views/clients/rollup/_assessments.html.haml]: Yes, much better thanks!.

**Comment by gigxz** [comment]
@eanders I had some other feedback but it felt easier to tinker with - see https://github.com/greenriver/hmis-warehouse/pull/4163!

**Review by gigxz — APPROVED** [review]
Looks great! Thank you!

**Pull request merged**